### PR TITLE
Ruby 3.2.2 upgrade

### PR DESCRIPTION
**Pull request by argvniyx-enroute**
/task https://app.asana.com/0/1204779908415024/1205454087538425/f

- bump `.ruby-version`
- bump `required_ruby_version` to be `>= 3.0` in gemspec
- bump `TargetRubyVersion` in `rubocop.yml` to be 3.0 
- regen the todo file

lmk if you want me to update the version and the changelog (held off because I saw other PR's don't necessarily do this)

@smudge @effron

### CODE REVIEW

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -49,6 +49,15 @@ Performance/MapMethodChain:
...
     - 'lib/rubocop/cop/betterment/server_error_assertion.rb'
 
+# Offense count: 3
+# Configuration parameters: Include, CustomTransform, IgnoreMethods, SpecSuffixOnly.
+# Include: **/*_spec*rb*, **/spec/**/*
+RSpec/FilePath:
+  Exclude:
+    - 'spec/rubocop/cop/betterment/utils/hardcoded_attribute_spec.rb'
+    - 'spec/rubocop/cop/betterment/utils/method_return_table_spec.rb'
+    - 'spec/rubocop/cop/betterment/utils/parser_spec.rb'
```
Comment (argvniyx-enroute) [inline .rubocop_todo.yml:59]: https://github.com/rubocop/rubocop-rspec/issues/1723

Comment (argvniyx-enroute) [reply .rubocop_todo.yml]: aaaaand it's gone

**Review by effron — APPROVED** [review]
domainLGTM && platformLGTM!

**Comment by dkubb** [comment]
__question:__ Should we also add `ruby file: '.ruby-version'` to the `Gemfile` in a follow-up?

**Comment by effron** [comment]
for posterity, we don't need to specify that here because this uses a `gemspec` to specify the required_ruby_version

**Pull request merged**